src/course-unit/move-modal/moveModal.test.tsx

```diff
@@ -0,0 +1,191 @@
+      ).toBeInTheDocument();
+      expect(within(breadcrumbs).getByText(sections[1].display_name)).toBeInTheDocument();
+      subsections.map((subsection) => (
+        expect(getByRole('button', { name: new RegExp(subsection.display_name, 'i') })).toBeInTheDocument()
+      ));
+    });
+
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(subsections[1].display_name, 'i') })));
+    await waitFor(() => {
+      expect(
+        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsUnits.defaultMessage),
+      ).toBeInTheDocument();
+      expect(within(breadcrumbs).getByText(subsections[1].display_name)).toBeInTheDocument();
+      units.map((unit) => (
+        expect(getByRole('button', { name: new RegExp(unit.display_name, 'i') })).toBeInTheDocument()
+      ));
+    });
+
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(units[0].display_name, 'i') })));
+    await waitFor(() => {
+      expect(
+        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsComponents.defaultMessage),
+      ).toBeInTheDocument();
+      expect(within(breadcrumbs).getByText(units[0].display_name)).toBeInTheDocument();
+      components.forEach((component) => {
+        if (component.display_name) {
+          expect(getByText(component.display_name)).toBeInTheDocument();
+        }
+      });
+    });
+  });
+
+  it('correctly navigates using breadcrumbs', async () => {
+    const { getByRole, getByTestId } = renderComponent();
+    const breadcrumbs: HTMLElement = getByTestId('move-xblock-modal-breadcrumbs');
+    const categoryIndicator: HTMLElement = getByTestId('move-xblock-modal-category');
+
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(sections[1].display_name, 'i') })));
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(subsections[1].display_name, 'i') })));
+    await waitFor(() => userEvent.click(within(breadcrumbs).getByText(sections[1].display_name)));
+
+    await waitFor(() => {
+      expect(
+        within(categoryIndicator).getByText(messages.moveModalBreadcrumbsSubsections.defaultMessage),
+      ).toBeInTheDocument();
+      expect(within(breadcrumbs).getByText(sections[1].display_name)).toBeInTheDocument();
+      subsections.map((subsection) => (
+        expect(getByRole('button', { name: new RegExp(subsection.display_name, 'i') })).toBeInTheDocument()
+      ));
+    });
+  });
+
+  it('renders empty message when no components are provided', async () => {
+    const { getByText, getByRole } = renderComponent();
+
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(sections[1].display_name, 'i') })));
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(subsections[1].display_name, 'i') })));
+    await waitFor(() => userEvent.click(getByRole('button', { name: new RegExp(units[7].display_name, 'i') })));
```

Comment:
[optional]: It seems to me that a wrapper of one `waitFor` will work the same as 3.

Reply:
Changed